appinventor/components/src/com/google/appinventor/components/runtime/util/JavaJoinListOfStrings.java

```diff
@@ -46,12 +46,22 @@ private static String join(List<Object> list, String separator)
   {
      StringBuilder sb = new StringBuilder();
      boolean first = true;
+     boolean firstSkip = false;
```

Comment:
Rather than modifying this function, you can wrap the list in a call to `yail-list-contents`, which gets you the tail of the list after the `*list*` symbol.

Reply:
I have reverted this .java file to the original .java file in master/ucr. I have edited runtime.scm by wrapping the list with yail-list-contents, seems to work properly!